📝(project) refacto documentation for Ralph V4 (#492)
openfun/ralph: docs/models/index.md

Author: About that, I am not sure if exposing the documentation of Pydantic models is necessary, as we used to do. I find it unreadable. What's your opinion on that ?

After structuring all the documentation, maybe I will move the path for some documentation, and this one could just become `docs/models.md`

> # Learning Statements Models

Reviewer: ```suggestion
# Learning Statement Models
```
Both are correct, but I find removing the "s" make this title slightly clearer (putting a bit more emphasis on "Models")

> The Learning Statements Models feature in Ralph empowers you to work with the LRS and ensure the quality of xAPI statements.

Reviewer: ```suggestion
The Learning Statement Models validation and conversion tools in Ralph empower you to work with the LRS and ensure the quality of xAPI statements.
```
?

> This feature not only enhances the integrity of your learning data but also facilitates integration and compliance with industry standards.

Reviewer: ```suggestion
These features not only enhance the integrity of your learning data but also facilitate integration and compliance with industry standards.
```

This section provides insights into the supported models, their conversion, and validation.

## Supported statements

> Our Learning Statements Models encompass a wide array of xAPI and OpenEdx statement types, ensuring comprehensive support for your e-learning data. Here are some of the key supported models:

Reviewer: ```suggestion
Our Learning Statements Models encompass a wide array of xAPI and OpenEdx statement types, ensuring comprehensive support for your e-learning data. Here are some of the key supported model categories:
```
?

1. **xAPI Statements Models**:
    - LMS
    - Video
    - Virtual classroom

2. **OpenEdx Statements Models**:
    - Enrollment
    - Navigational
    - Open Reponse Assessment
    - Peer instruction
    - Problem interaction
    - Textbook interaction
    - Video

## Statements validation

> The validation statements feature allows you to verify the validity of supported statements. This is crucial for maintaining data accuracy and compliance with industry standards. The validation will notify you of any discrepancies or issues in your statement(s).

Reviewer: ```suggestion
The statements validation feature allows you to verify the validity of supported statements. This is crucial for maintaining data accuracy and compliance with industry standards. The validation will notify you of any discrepancies or issues in your statement(s).
```

> The validation statements feature allows you to verify the validity of supported statements. This is crucial for maintaining data accuracy and compliance with industry standards. The validation will notify you of any discrepancies or issues in your statement(s).

Reviewer: What types of "discrepancies" does Ralph cover ?

Check out our tutorials to test the validation feature:

- [`validate` with CLI](../tutorials/cli.md#validate-command)

> - `validate` with `ralph-malph` Pypi package

Author: I have changed to `validate with Ralph as a library`. My initial formulation was incorrect, as we can use the CLI from the pypi package. There was some confusion

## [WIP] Statements conversion

> The statements conversion feature offers a seamless transition between OpenEdx and xAPI statement formats, promoting interoperability and adaptability. By converting OpenEdx statements to xAPI, you ensure that your learning data can be easily integrated into various e-learning ecosystems.

Reviewer: ```suggestion
The statements conversion feature offers a seamless transition between OpenEdx and xAPI statement formats, promoting interoperability. By converting OpenEdx statements to xAPI, you ensure that your learning data can be easily integrated into various e-learning ecosystems.
```

# Learning statement models

The learning statement models validation and conversion tools in Ralph empower you to work with an LRS and ensure the quality of xAPI statements.

> These features not only enhance the integrity of your learning data but also facilitates integration and compliance with industry standards.

Reviewer: ```suggestion
These features not only enhance the integrity of your learning data but also facilitate integration and compliance with industry standards.
```

## Statements validation

In learning analytics, the validation of statements takes on significant importance. 
These statements, originating from diverse sources, systems or applications, must align with specific standards such as [xAPI](https://xapi.com/) for the best known. 
The validation process becomes essential in ensuring that these statements meet the required standards, facilitating data quality and reliability. 

Ralph allows you to automate the validation process in your production stack. 
OpenEdx related events and xAPI statements are supported.

!!! warning 

    For now, validation is effective only with supported [learning statement models](#learning-statement-models) on Ralph. About xAPI statements, an [issue](https://github.com/openfun/ralph/issues/388) is open to extend validation to any xAPI statement.

Check out tutorials to test the validation feature:

- [`validate` with Ralph as a CLI](../tutorials/cli.md#validate-command)
- [`validate` with Ralph as a library](../tutorials/library.md#validate-method)

> ## Statements conversion

Reviewer 2: :1st_place_medal:

Ralph currently supports conversion from OpenEdx learning events to xAPI statements. Here is the up-to-date conversion sets availables: 

| FROM | TO |
|---|---|

> |[edx.course.enrollment.activated](https://edx.readthedocs.io/projects/devdata/en/latest/internal_data_formats/tracking_logs/student_event_types.html#edx-course-enrollment-activated-and-edx-course-enrollment-deactivated)|[registered to a course](https://profiles.adlnet.gov/profile/d0f8a66c-df56-4cb4-bfaf-10c2febd87a1/templates/685a9da1-0d8c-4ae5-82bf-fff7b601c288)|

Reviewer 2: What about using reference-style markdown links here? (as we do in the changelog for example)

# Learning statement models

The learning statement models validation and conversion tools in Ralph empower you to work with an LRS and ensure the quality of xAPI statements.

> This features not only enhance the integrity of your learning data but also facilitates integration and compliance with industry standards.

Reviewer: ```suggestion
This features not only enhances the integrity of your learning data but also facilitates integration and compliance with industry standards.
```

    - [Navigational](https://edx.readthedocs.io/projects/devdata/en/latest/internal_data_formats/tracking_logs/student_event_types.html#course-navigation-events)
    - [Open Reponse Assessment](https://edx.readthedocs.io/projects/devdata/en/latest/internal_data_formats/tracking_logs/student_event_types.html#open-response-assessment-events)
    - [Peer instruction](https://edx.readthedocs.io/projects/devdata/en/latest/internal_data_formats/tracking_logs/student_event_types.html#peer-instruction-events)
    - [Problem interaction](https://edx.readthedocs.io/projects/devdata/en/latest/internal_data_formats/tracking_logs/student_event_types.html#problem-interaction-events)
    - [Textbook interaction](https://edx.readthedocs.io/projects/devdata/en/latest/internal_data_formats/tracking_logs/student_event_types.html#textbook-interaction-events)
    - [Video interaction](https://edx.readthedocs.io/projects/devdata/en/latest/internal_data_formats/tracking_logs/student_event_types.html#video-interaction-events)

## Statements validation

In learning analytics, the validation of statements takes on significant importance. 

> These statements, originating from diverse sources, systems or applications, must align with specific standards such as [xAPI](https://xapi.com/) or [Caliper](https://www.imsglobal.org/activity/caliper), for the best known. 

Reviewer: Do we support Caliper ? If not, I would leave it out to avoid confusing the reader.

The validation process becomes essential in ensuring that these statements meet the required standards, facilitating data quality and reliability. 

> Ralph proposes a validation statements tool allowing you to automate the validation process in your production stack. 

Reviewer: ```suggestion
Ralph allows you to automate the validation process in your production stack. 
```

> OpenEdx related events and xAPI statements are supported in the project. 

Reviewer: ```suggestion
OpenEdx related events and xAPI statements are supported. 
```

> Based on Pydantic modelisation for the set of statements supported by Ralph, the validation feature provides a streamlined, reliable, and efficient approach to validate large volumes of learning statements.

Reviewer: ```suggestion
```
This is mostly said above (except for Pydantic, which doesn't seem necessary to mention).

!!! warning 

    For now, validation is effective only with supported [learning statement models](#learning-statement-models) on Ralph. About xAPI statements, an [issue](https://github.com/openfun/ralph/issues/388) is open to extend validation to any xAPI statement.

Check out tutorials to test the validation feature:

- [`validate` with Ralph as a CLI](../tutorials/cli.md#validate-command)
- [`validate` with Ralph as a library](../tutorials/library.md#validate-method)

## Statements conversion

> The statements conversion feature offers a seamless transition between OpenEdx and xAPI statement formats, promoting interoperability and adaptability. 
> 
> By converting OpenEdx statements to xAPI, you ensure that your learning data can be easily integrated into various e-learning ecosystems.

Reviewer: ```suggestion
```
In my opinion, the typical reader will not need to be convinced that conversion is a good idea (either they need it or they don't)
